josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -615,6 +627,15 @@ async fn call_service(
         }
     }
 
+    let span = tracing::span!(tracing::Level::TRACE, "hyper_cgi");
+    let _enter = span.enter();
+    let mut context_propagator = HashMap::<String, String>::default();
+    let context = span.context();
+    global::get_text_map_propagator(|propagator| {
+        propagator.inject_context(&context, &mut context_propagator);
+    });
```

Comment:
does `global` mean that global program state is being manipulated here? would be helpful if there's a comment about that

Reply:
Here we're only reading from the global configuration. This is commented on the writing side

Reply:
sorry, not the writing side, but the part where I set it up. This is the part where we write in it :D